include/graph.h

```diff
@@ -0,0 +1,44 @@
+#pragma once
+#include <cstdlib>
+#include <exception>
+#include <set>
+#include "supernode.h"
+
+typedef unsigned long long int Node;
+typedef std::pair<Node, Node> Edge;
+
+enum UpdateType {
+  INSERT,
+  DELETE,
+};
+
+typedef pair<Edge, UpdateType> GraphUpdate;
+
+class Graph{
+  const unsigned long long int num_nodes;
+  bool UPDATE_LOCKED = false;

```

Comment:
nit: I don't think this has to be all-caps. Nothing else is and I think it's at best misleading (it isn't const)